src/System.Data.SqlClient/tests/FunctionalTests/CloneTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Data.SqlClient.Tests
+{
+    public class CloneTests
+    {
+        [Fact]
+        public void CloneSqlConnection()
+        {
+            SqlConnectionStringBuilder builder = new SqlConnectionStringBuilder();
+            builder.DataSource = "mybadServer";
+            builder.ConnectTimeout = 1;
+            builder.InitialCatalog = "northwinddb";
+            SqlConnection connection = new SqlConnection(builder.ConnectionString);
+
+            SqlConnection clonedConnection = (connection as ICloneable).Clone() as SqlConnection;
+            Assert.Equal(connection.ConnectionString, clonedConnection.ConnectionString);
+            Assert.Equal(connection.ConnectionTimeout, clonedConnection.ConnectionTimeout);
+            Assert.NotEqual(connection, clonedConnection);
+        }
+
+        [Fact]
+        public void CloneSqlCommand()
+        {
+            SqlConnection connection = new SqlConnection();
+
+            SqlCommand command = connection.CreateCommand();
+            command.CommandText = "select 1";
+            command.CommandTimeout = 45;
+            command.CommandType = CommandType.Text;
+
+            SqlParameter parameter = command.CreateParameter();
+            parameter.Direction = ParameterDirection.Input;
+            parameter.DbType = DbType.Currency;
+            parameter.Size = 10;
+            parameter.Precision = 3;
+            parameter.Scale = 5;
+
+            command.Parameters.Add(parameter);
+
+            SqlCommand clonedCommand = command.Clone();
+            compareCommands(command, clonedCommand);
+
+            SqlCommand clonedByICloneable = (command as ICloneable).Clone() as SqlCommand;
```

Comment:
Same as the reasoning for `SqlConnection` Clone()
For SqlCommand, I am testing the API SqlCommand.Clone and ICloneable.Clone() and hence there is no casting in one of the Clones and there is a need to cast in the `ICloneable.Clone()` call